lib/datadog/tracing/contrib/delayed_job/configuration/settings.rb

```diff
@@ -13,20 +13,29 @@ module Configuration
           # @public_api
           class Settings < Contrib::Configuration::Settings
             option :enabled do |o|
-              o.default { env_to_bool(Ext::ENV_ENABLED, true) }
+              o.type :bool
+              o.env Ext::ENV_ENABLED
+              o.default true
             end
 
             option :analytics_enabled do |o|
-              o.default { env_to_bool(Ext::ENV_ANALYTICS_ENABLED, false) }
+              o.type :bool
+              o.env Ext::ENV_ANALYTICS_ENABLED
+              o.default false
             end
 
             option :analytics_sample_rate do |o|
-              o.default { env_to_float(Ext::ENV_ANALYTICS_SAMPLE_RATE, 1.0) }
+              o.type :float
+              o.env Ext::ENV_ANALYTICS_SAMPLE_RATE
+              o.default 1.0
             end
 
             option :service_name
             option :client_service_name
-            option :error_handler, experimental_default_proc: Tracing::SpanOperation::Events::DEFAULT_ON_ERROR
+            option :error_handler do |o|
+              o.type :proc
+              o.experimental_default_proc(&Tracing::SpanOperation::Events::DEFAULT_ON_ERROR)
```

Comment:
Not for this PR -- but are we're ready to make a decision on the `experimental_default_proc`? 

E.g. do we want to keep it as-is for now (probably rename it), or do you still have plans to change/remove it?

Reply:
For now, no plans to remove it. We should rename it to `default_proc` 😄

Reply:
Ack! Created https://github.com/DataDog/ruby-guild/issues/29 to record it (Datadog-only link)